### DISCUSSION TITLE
Is there a tool to calculate Rails app configuration for maximum performance based on given inputs?

### DISCUSSION THREAD

**Discussion by dhavalsingh** [puma/puma#3222] [Q&A]
I came across https://pgtune.leopard.in.ua/ today. I want to build (if it doesn’t exist already) something like this for Rails app.

For eg: User inputs RAM, CPU, Server(if Puma then single or cluster mode), Type of application, and a few other things.. Maybe once these things are set, in the future integrate K8s with Requests & Limits, Replica counts, and such.

I know all these differ a lot based on each use case but the idea is to give the user a good starting point. The logic will be simple based on formulas, nothing too fancy, mostly a static website.

Is this possible? What do you guys think?

**Comment by nateberkopec** [comment]
Somebody actually showed me a demo of this once and now I can't find it again 😢 

Yeah, that could probably be useful!

**Comment by nateberkopec** [comment]
For example, we could suggest different configs based on if you're using a reverse proxy (NGINX, Apache) or not.

**Comment by dhavalsingh** [comment]
Yeah, if anyone has any resources/ideas please feel free to share. I will start working on this soon once I get few things organised. If anyone wants to do this lmk, open to collaboration
